enterprise/frontend/src/embedding-sdk-bundle/components/public/MetabotQuestion/MetabotQuestion.tsx

```diff
@@ -7,35 +7,39 @@ import {
 } from "embedding-sdk-bundle/components/private/PublicComponentWrapper";
 import { SdkAdHocQuestion } from "embedding-sdk-bundle/components/private/SdkAdHocQuestion";
 import { SdkQuestionDefaultView } from "embedding-sdk-bundle/components/private/SdkQuestionDefaultView";
+import { useSdkDispatch } from "embedding-sdk-bundle/store";
 import { useLocale } from "metabase/common/hooks/use-locale";
-import { Flex, Icon, Paper, Stack, Text } from "metabase/ui";
-import { METABOT_ERR_MSG } from "metabase-enterprise/metabot/constants";
+import { Flex, Paper, Stack, Text } from "metabase/ui";
+import { Messages } from "metabase-enterprise/metabot/components/MetabotChat/MetabotChatMessage";
+import { useMetabotAgent } from "metabase-enterprise/metabot/hooks";
+import { addNavigateToHandler } from "metabase-enterprise/metabot/state";
 
 import { MetabotChatEmbedding } from "./MetabotChatEmbedding";
 import { QuestionDetails } from "./QuestionDetails";
 import { QuestionTitle } from "./QuestionTitle";
 
 const MetabotQuestionInner = () => {
   const { isLocaleLoading } = useLocale();
-  const [redirectUrl, setRedirectUrl] = useState<string>("");
-  const [messages, setMessages] = useState<string[]>([]);
+  const [adHocQuestionUrl, setAdHocQuestionUrl] = useState<string>("");
+  const dispatch = useSdkDispatch();
+
+  useEffect(() => {
+    dispatch(addNavigateToHandler(setAdHocQuestionUrl));
```

Comment:
I assume we don't support multiple `<MetabotQuestion>` components on a page, so we set it to the store